### PR TITLE
feat: add ApiClient#registerEvents

### PR DESCRIPTION
**Pull request by yshrsmz**
todo

- [x] https://github.com/bucketeer-io/android-client-sdk/issues/56

### CODE REVIEW

File: src/internal/model/Event.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  tag: string
+  sourceId: SourceID
+  sdkVersion: string
+  metadata: Record<string, string>
+}
+
+export type MetricsEvent =
+  | {
+      timestamp: number
+      type: typeof MetricsEventType['GET_EVALUATION_LATENCY']
```
[the remark this answers is not in the document]
Comment (yshrsmz) [reply src/internal/model/Event.ts:49]: (current code is based on the previous one, so will update once I get the answer)

[the remark this answers is not in the document]
Comment (cre8ivejp) [reply src/internal/model/Event.ts:49]: We recently updated the node js server sdk. This PR implements the new metrics, including the `@type` needed in the request.

https://github.com/ca-dp/bucketeer-node-server-sdk/blob/master/src/objects/metricsEvent.ts
https://github.com/ca-dp/bucketeer-node-server-sdk/blob/master/src/objects/status.ts

You can the implementation in this PR.
https://github.com/ca-dp/bucketeer-node-server-sdk/pull/14

Please let me know if it clarifies for you.

[the remark this answers is not in the document]
Comment (yshrsmz) [reply src/internal/model/Event.ts]: thanks, updated in [`180fb60` (#17)](https://github.com/bucketeer-io/javascript-client-sdk/pull/17/commits/180fb60d36a1a0dc9406f04202408a438c44401e)

**Comment by yshrsmz** [comment]
@cre8ivejp should be ready now. please take a look :pray:

File: src/internal/model/MetricsEventData.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  '@type': typeof MetricsEventType['ServiceUnavailableError']
+}
+
+export interface TimeoutErrorMetricsEvent {
+  apiId: ApiId
+  labels: Record<string, string>
+  '@type': typeof MetricsEventType['TimeoutError']
+}
+
+export interface InternalErrorMetricsEvent {
```
Comment (cre8ivejp) [inline src/internal/model/MetricsEventData.ts:94]: This `InternalErrorMetricsEvent` is deprecated. 
We can exclude it from the event list.

File: src/internal/model/response/RegisterEventsErrorResponse.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export interface RegisterEventsErrorResponse {
+  retrieable: boolean
```
Comment (cre8ivejp) [inline src/internal/model/response/RegisterEventsErrorResponse.ts:2]: nit: `retriable`

File: src/internal/model/MetricsEventData.ts (hunk 2 above)
Comment (yshrsmz) [reply src/internal/model/MetricsEventData.ts]: thanks!

[`7004211` (#17)](https://github.com/bucketeer-io/javascript-client-sdk/pull/17/commits/700421107c7421f925290adbacac77695f580421)

File: src/internal/model/response/RegisterEventsErrorResponse.ts (hunk 3 above)
Comment (yshrsmz) [reply src/internal/model/response/RegisterEventsErrorResponse.ts]: [`43a9264` (#17)](https://github.com/bucketeer-io/javascript-client-sdk/pull/17/commits/43a92644b85e4ed4518c398f359543d123cdb847)

**Review by cre8ivejp — APPROVED** [review]
Awesome!
Thank you!

**Pull request merged**